### PR TITLE
feat: add all round happy path

### PR DESCRIPTION
**Pull request by rkolpakov**

### CODE REVIEW

File: tests/regression/test_all_round_happy_path.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    assert almostEqEth(transfer_shares_event["sharesValue"], shares_to_be_minted)
+
+    assert total_supply_after_submit == total_supply_before_submit + amount
+    assert buffered_ether_after_submit == buffered_ether_before_submit + amount
+    assert staking_limit_after_submit == staking_limit_before_submit - amount
+
+    # Depositing ETH
+    dsm = accounts.at(contracts.deposit_security_module.address, force=True)
+
+    contracts.lido.deposit(max_deposit, curated_module_id, "0x0", {"from": dsm})
```
Comment (TheDZhon) [inline tests/regression/test_all_round_happy_path.py:55]: can add a check for `getDepositableEther` (should be equal to `getBufferedEther` when no withdrawal request placed)

File: tests/regression/test_all_round_happy_path.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    # Rebasing (Increasing balance)
+
+    report_tx, _ = oracle_report(cl_diff=ETH(100), exclude_vaults_balances=True)
+    steth_balance_after_rebase = contracts.lido.balanceOf(stranger)
+
+    token_rebased_event = report_tx.events["TokenRebased"]
+
+    assert report_tx.events.count("TokenRebased") == 1
+    assert report_tx.events.count("WithdrawalsFinalized") == 0
+    assert token_rebased_event["postTotalEther"] - token_rebased_event["preTotalEther"] == amount
```
Comment (TheDZhon) [inline tests/regression/test_all_round_happy_path.py:69]: can check also:
```python
assert token_rebased_event["postTotalShares"] == token_rebased_event["preTotalShares"] + token_rebased_event["sharesMintedAsFees"]
```

File: tests/regression/test_all_round_happy_path.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    [(_, _, _, _, finalized, _)] = contracts.withdrawal_queue.getWithdrawalStatus(request_ids)
+
+    assert almostEqEth(steth_balance_after_withdrawal_request, steth_balance_after_rebase - amount)
+    assert len(contracts.withdrawal_queue.getWithdrawalRequests(stranger, {"from": stranger})) == 1
+    assert contracts.withdrawal_queue.getLastRequestId() == 1
+    assert not finalized
+
+    # Rebasing (Withdrawal finalization)
+
+    locked_ether_amount_before_finalization = contracts.withdrawal_queue.getLockedEtherAmount()
```
Comment (TheDZhon) [inline tests/regression/test_all_round_happy_path.py:120]: can add a check:
```python
assert contracts.lido.balanceOf(contracts.withdrawal_queue) == amount
# ... finalization ...
assert contracts.lido.balanceOf(contracts.withdrawal_queue) == 0
```

**Review by TheDZhon — COMMENTED** [review]
LGTM, please consider adding some additional (paranoid-mode) checks :pray:

File: tests/regression/test_all_round_happy_path.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    dsm = accounts.at(contracts.deposit_security_module.address, force=True)
+
+    contracts.lido.deposit(max_deposit, curated_module_id, "0x0", {"from": dsm})
+    buffered_ether_after_deposit = contracts.lido.getBufferedEther()
+
+    assert buffered_ether_after_submit > buffered_ether_after_deposit
+
+    # Rebasing (Increasing balance)
+
+    report_tx, _ = oracle_report(cl_diff=ETH(100), exclude_vaults_balances=True)
```
Comment (folkyatina) [inline tests/regression/test_all_round_happy_path.py:62]: Why do we excluding vault balances here? It does not suppose to happen during the happy path, imho, doesn't it?

File: tests/regression/test_all_round_happy_path.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    steth_balance_after_rebase = contracts.lido.balanceOf(stranger)
+
+    token_rebased_event = report_tx.events["TokenRebased"]
+
+    assert report_tx.events.count("TokenRebased") == 1
+    assert report_tx.events.count("WithdrawalsFinalized") == 0
+    assert token_rebased_event["postTotalEther"] - token_rebased_event["preTotalEther"] == amount
+
+    assert steth_balance_after_rebase > steth_balance_after_submit
```
Comment (folkyatina) [inline tests/regression/test_all_round_happy_path.py:71]: Should we check here that some rewards landed to NOs and some - to treasury?

File: tests/regression/test_all_round_happy_path.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+    # Requesting withdrawal
+
+    assert len(contracts.withdrawal_queue.getWithdrawalRequests(stranger, {"from": stranger})) == 0
+
+    approve_tx = contracts.lido.approve(
+        contracts.withdrawal_queue.address, amount, {"from": stranger}
```
Comment (folkyatina) [inline tests/regression/test_all_round_happy_path.py:78]: IMHO, should check that it's possible to withdraw increased amount here, but not an original one.

**Review by folkyatina — APPROVED** [review]
It's great, but can be extended a bit.

**Pull request merged**